src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -675,10 +667,13 @@ private void InitLocalCultureInfo()
             Stloc(_cultureV);
         }
 
+        /// <summary>Whether ToLower operations should be performed with the invariant culture as opposed to the one in <see cref="_cultureV"/>.</summary>
+        private bool UseToLowerInvariant => _cultureV == null || _options.HasFlag(RegexOptions.CultureInvariant);
```

Comment:
HasFlag is intrinsic these days, fast enough you don't need to cache this?

Reply:
Yes:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgBUYEMWAlGAcwCuAG2xQAoggAOsXLgCWEAHa4A3DRrEAzE1IMAwgwDeNBmaY7gECMIYBZcgAo+/TgHkpGRSoYRP33ABKBgBeAD5ff2VcFgAJPAAxUX5nAXcolRZ9EQxBWABJJQA3MXlsJQxA9WpzCwYrG3tSVNcEDy9oyI6VYPCGRz9u3AYAMgYXdKGsnLyYQpKoMorggEIQhjpqgF8gA
But I could be explicit about it if you'd prefer.